### PR TITLE
Default login

### PR DESCRIPTION
**Pull request by sblack-usu**
I'll push this out shortly after 1.16.1 completes.  It adds referenced content methods and also documents overwrite on unzip.  I also snuck in user prompt if no authentication is provided.

### CODE REVIEW

File: hs_restclient/__init__.py [hunk 1]
```diff
@@ -1075,6 +1082,66 @@ def deleteResourceFolder(self, pid, pathname):
 
         return r.json()
 
+    def createReferencedFile(self, pid, path, name, ref_url):
+        """Delete a folder as specified by *pathname* for a given resource
```
Comment (pkdash) [inline hs_restclient/__init__.py]: This doc string description of the functionality does not seem right.

File: hs_restclient/__init__.py [hunk 2]
```diff
@@ -1075,6 +1082,66 @@ def deleteResourceFolder(self, pid, pathname):
...
+                raise HydroShareNotAuthorized(('POST', url))
+            elif r.status_code == 404:
+                raise HydroShareNotFound((pid,))
+            else:
+                raise HydroShareHTTPException((url, 'POST', r.status_code))
+
+        return r.json()
+
+    def updateReferencedFile(self, pid, path, name, ref_url):
+        """Delete a folder as specified by *pathname* for a given resource
```
Comment (pkdash) [inline hs_restclient/__init__.py]: Functionality description seems not right.

File: hs_restclient/__init__.py [hunk 3]
```diff
@@ -1075,6 +1082,66 @@ def deleteResourceFolder(self, pid, pathname):
...
+
+                :param pid: The HydroShare ID of the resource for which the file should be updated
+                :param path: Folder path for the file to be updated in
+                :param name: Filename for the referenced file
+                :param ref_url: url to be used in the referenced file
```
Comment (pkdash) [inline hs_restclient/__init__.py]: May be this change is better.
'url to be used in the referenced file' -> 'url to be updated in the referenced file'

File: tests/test_hs_restclient.py [hunk 4]
```diff
@@ -470,5 +470,25 @@ def test_resource_list_by_bounding_box(self):
...
+                                           ref_url='https://www.hydroshare.org')
+
+        self.assertEqual(response['status'], 'success')
+
+    @with_httmock(mocks.hydroshare.resourceUpdateReferenceURL_post)
+    def test_referenced_update(self):
+        hs = HydroShare(prompt_auth=False)
+
+        response = hs.updateReferencedFile(pid='511debf8858a4ea081f78d66870da76c', path='data/contents', name='file.url',
+                                           ref_url='https://www.hydroshare.org')
```
Comment (pkdash) [inline tests/test_hs_restclient.py]: Should we use a different value of the ref_url than what has been used in creating the referenced file?

**Comment by pkdash** [comment]
@sblack-usu Left few comments for you to consider. Thanks.

**Comment by sblack-usu** [comment]
Thanks @pkdash, all good feedback.  I've updated the branch with the changes you requested.

**Pull request merged**